core/src/main/java/org/apache/iceberg/BaseIncrementalChangelogScan.java

```diff
@@ -153,6 +156,9 @@ public CloseableIterable<ChangelogScanTask> apply(
             int changeOrdinal = snapshotOrdinals.get(commitSnapshotId);
             DataFile dataFile = entry.file().copy(context.shouldKeepStats());
 
+            context.scanMetrics().resultDataFiles().increment();
```

Comment:
I agree this isn't accurate atm because we currently don't support dimensional counters (and I should have added a TODO/mention here). I think we have two options to deal with this:

1) we could add `resultDataFilesByAddedRowsScanTask()` /  `resultDataFilesByDeletedDataFilesScanTask()` to `ScanMetrics` / `ScanMetricsResult` for this particular case. This would mean that we'd have to support this for at least a minor release.
2) we implement dimensional counters/timers (which we will need anyway in the long run) and mention that this part is currently not accurate.

Thoughts?

Reply:
Dimensional counters might be helpful, but really I think the need is to break down by task type. I don't know that we want to consider these the same counter.